app/jobs/course/assessment/answer/auto_grading_job.rb

```diff
@@ -41,17 +43,30 @@ class Course::Assessment::Answer::AutoGradingJob < ApplicationJob
   # @param [String] redirect_to_path The path to redirect when job finishes.
   def perform_tracked(answer, redirect_to_path = nil)
     ActsAsTenant.without_tenant do
+      raise PriorityShouldBeLoweredError if queue_name == 'highest' && answer.question.is_low_priority
+
+      start_time = Time.now
       Course::Assessment::Answer::AutoGradingService.grade(answer)
       if update_exp?(answer.submission)
         Course::Assessment::Submission::CalculateExpService.update_exp(answer.submission)
       end
+
+      if queue_name == 'highest' && (Time.now - start_time > answer.auto_grading_time_limit)
```

Comment:
nit: personally I would check if the question is high priority, and taking a long time, then lower priority:
e.g. `if !answer.question.is_low_priority && (Time.now - start_time > answer.auto_grading_time_limit)`

Reply:
done